DOCS-2541: Move component APIs (#3504)
viamrobotics/docs: docs/components/input-controller/_index.md

Reviewer: Hmm it's weird that GH just shows red boxes on the top bar here as though this file is just being straight-up deleted, even though it shows the diff correctly.

# SME: James
---

Input controllers provide an API for configuring callbacks for events, allowing you to configure input devices to control your machines.

> If you have a keyboards, mice, elevator button panels, light power switches, joysticks, and gamepads, or, video game controllers with which you want to control a robotic base, use an _input controller_ component.

Reviewer: ```suggestion
If you have a keyboard, mouse, elevator button panel, light power switch, joystick, gamepad, or video game controllers with which you want to control a robotic base, use an _input controller_ component.
```

This component supports devices like gamepads and joysticks that contain one or more [`Control`s](/appendix/apis/components/input-controller/#control-field) representing the individual axes and buttons on the device.
To use the controller's inputs, you must [register callback functions](/appendix/apis/components/input-controller/#registercontrolcallback) to the [`Control`s](/appendix/apis/components/input-controller/#control-field) with the [`input` API](/appendix/apis/components/input-controller/).

The callback functions can then handle the [Events](/appendix/apis/components/input-controller/#getevents) that are sent when the `Control` is activated or moved.
For example, when a specific button is pushed, the callback function registered to it can move another component, or print a specific output.

> The [base remote control service](/services/base-rc/) implements an input controller as a remote control for a base.

Reviewer: I think this is a bit vague/not helpful, though of course they can click the link.

Author: you're right but I will leave that for when we review the page content more thoroughly.

## Available models

To use an input controller to control your machine's actions, you need to add it to your machine's configuration.

Go to your machine's **CONFIGURE** page, and add a model that supports your input controller.

> The following list shows you the available input controller models.

Reviewer: ```suggestion
The following list shows the available input controller models.
```

{{% /tab %}}
{{< /tabs >}}

## API

> The [input controller API](/appendix/apis/components/input-controller/), which supports the following methods:

Reviewer: ```suggestion
The [input controller API](/appendix/apis/components/input-controller/) supports the following methods:
```
also apply to other components
